### PR TITLE
CA-4329 Add order transaction retrieval with apple pay fields on receipts and update marshaler to jackson

### PR DESCRIPTION
**Pull request by ryankazokas**
* Shopfiy SDK will now support Apple Pay transaction receipts
* Objects will be marshaled via Jackson now.
* Inventory Quantity is made read only be removing setter

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -143,7 +148,7 @@
 							<goal>mutationCoverage</goal>
 						</goals>
 						<configuration>
-							<mutationThreshold>70</mutationThreshold>
+							<mutationThreshold>81</mutationThreshold>
```
Comment (rjdavis3) [inline pom.xml]: Nice increase!

File: src/main/java/com/shopify/model/adapters/EscapedStringsAdapter.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 	@Override
 	public Collection<String> unmarshal(final Collection<String> escapedStrings) throws Exception {
 		return (escapedStrings == null) ? null
-				: escapedStrings.stream().map(StringEscapeUtils::unescapeHtml4).collect(Collectors.toSet());
+				: escapedStrings.stream().map(StringEscapeUtils::unescapeHtml4).collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (ryankazokas) [reply src/main/java/com/shopify/model/adapters/EscapedStringsAdapter.java]: No, I was getting an error from jackson, that these specificially needed to be a list.

File: src/main/java/com/shopify/model/ShopifyTransactionReceipt.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+@XmlRootElement
+@XmlAccessorType(XmlAccessType.FIELD)
+public class ShopifyTransactionReceipt {
+
+	@XmlElement(name = "apple_pay")
```
[the remark this answers is not in the document]
Comment (ryankazokas) [reply src/main/java/com/shopify/model/ShopifyTransactionReceipt.java]: Yeah i agree. I think this upfront work will make things easier in the future. Especially if we need to map any more dynamic objects.

**Comment by ryankazokas** [comment]
@rjdavis3 please re-review when you get a chance.

**Pull request merged**